src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -376,10 +377,18 @@ class PrisonerIepLevelReviewService(
       log.info("No incentive records found for $removedPrisonerNumber, no records updated")
     }
   }
-}
 
-@OptIn(FlowPreview::class)
-fun <T> merge(vararg flows: Flow<T>): Flow<T> = flowOf(*flows).flattenMerge()
+  suspend fun countOfPrisonersOnLevelInPrison(prisonId: String, levelCode: String): Int {
+    return offenderSearchService.findOffendersAtLocation(prisonId, prisonId)
+      .fold(0) { countSoFar, offenders ->
```

Comment:
🤔 I wonder if you could change the method name to something like `anyPrisonerOnLevelInPrison()` (possibly a better name) and shortcut/return true this whenever you found a match...

this could save some HTTP requests/queries/iterations with a bit of luck as you only interested in knowing if there is any prisoner at that level to prevent the Prison Incentive level deactivation right?

Reply:
i'm expecting to need the number returned for a future error message; if not then i agree with you! we can then use more performant SQL too

Reply:
@xoen actually, would you rather this method returned a boolean until we know that we really do want a number?